module-bsp/board/rt1051/bellpx/audio.cpp

```diff
@@ -16,10 +16,45 @@ using namespace drivers;
 
 AudioConfig audioConfig;
 
-void bsp::audio::init()
+namespace
+{
+    constexpr bool isDerivativeOf44100(const std::uint32_t sampleRate)
+    {
+        return sampleRate % 11025 == 0;
+    }
+
+    DriverPLLParams getPLLParams(const std::uint32_t sampleRate)
+    {
+        /// The basic formula for calculating PLL4:
+        /// (fsclk * (loopDivider + numerator/denominator)) / postDivider
+
+        if (isDerivativeOf44100(sampleRate)) {
+            /// PLL4 set to 722.5344MHz, later it will be divided by SAI to generate 11,2896MHz BCLK
+            return {
+                .loopDivider = 30,
+                .postDivider = 1,
+                .numerator   = 1056,
+                .denominator = 10000,
+                .src         = 0,
+            };
+        }
+        else {
+            /// PLL4 set to 786.48MHz, later it will be divided by SAI to generate 12,28875MHz BCLK
+            return {
+                .loopDivider = 32,
+                .postDivider = 1,
+                .numerator   = 77,
+                .denominator = 100,
+                .src         = 0,
+            };
```

Comment:
do we need to provide in bell codec different clock for different sample rates no internal PLL in codec?

Why this is needed at all? is there assumption to support wide range o sample rates?

Reply:
There is an internal PLL inside the codec, but it's not as powerful as the pure one. It can't generate all available sample rates from one MCLK hence it is required to provide different MCLK for a family of 44.1kHz sample rates and a different one for 48kHz.

Reply:
got it, thanks